hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto

```diff
@@ -754,7 +754,7 @@ message BucketInfo {
     required string bucketName = 2;
     repeated OzoneAclInfo acls = 3;
     required bool isVersionEnabled = 4 [default = false];
-    required StorageTypeProto storageType = 5 [default = DISK];
+    optional StorageTypeProto storageType = 5 [default = DISK, deprecated = true];
```

Comment:
is this change from required to optional backward compatible?
Imagine a new OM server and an old client. The new server omits this field while client expects this field. Wouldn't the client fail to parse?